### ISSUE TITLE
Option to rerun only requested tasks

### ISSUE THREAD

**Issue by davidburstromspotify** [gradle/gradle#20641]
### Expected Behavior

As a developer or DPE engineer, I want to rerun a given task, regardless if it's up to date or is cacheable. Typical scenarios are for benchmarking compilation tasks, debugging flaky tests, refreshing broken cache artifacts, etc.
I would like a flag (or equivalent) allowing me to run `./gradlew :path:to:my:task --rerun-requested-tasks`. The applicable task(s) should be subject to the regular Gradle task matching and abbreviations, so that `tas` would match all `task` instances in all included projects.

### Current Behavior

Using `--rerun-tasks` runs all the tasks up to and including the requested tasks, which wastes time and resources.

The closest approximation of the requested feature is to invoke `./gradlew :path:to:my:cleanTask :path:to:my:task --no-build-cache` which is both tedious and error-prone.

### Context

I have a need for this feature on at least a weekly basis, based on the given use-cases.

When a developer in a large repo has broken cache artifacts for a given task, it's performance prohibitive to tell the developer to run with `--rerun-tasks`, but it's also inconvenient to tell them to use `./gradlew :path:to:my:cleanTask :path:to:my:task --no-build-cache`, as the cache artifacts won't be refreshed.

**Comment by Kircheisser** [comment]
I agree this would be nice, where I work I added a property which is a comma separated list of tasks, and there's some configuration that marks the output of those tasks as upToDateWhen { false }.

This has the effect of the cache not being used, and the new cache entry being pushed.

That being said, if you need this feature on a weekly basis, maybe you should identify where your broken caches are coming from. Though I appreciate sometimes resolving it is out of your hands, e.g. if a plugin is broken.

**Comment by davidburstromspotify** [comment]
Oh, my "weekly basis" is usually not due to broken cache artefacts. Benchmarking and rerunning tests are more common. I agree that addressing cache problems should be a separate activity. It's usually related to underspecified inputs, orphaned output files or out-of-order task execution.
